src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -150,14 +157,32 @@ private String resolveHeadSha() {
                             getRepository(), head.get().getName()));
         }
 
-        if (revision.get() instanceof AbstractGitSCMSource.SCMRevisionImpl) {
-            return ((AbstractGitSCMSource.SCMRevisionImpl) revision.get()).getHash();
+        return resolveCommit(revision.get());
+    }
+
+    private String resolveCommit() {
+        if (run == null) {
+            return StringUtils.EMPTY;
+        }
+
+        Optional<SCMRevision> revision = scmFacade.findRevision(resolveSource(), run);
+        if (revision.isPresent()) {
+            return resolveCommit(revision.get());
+        }
+        else {
+            return StringUtils.EMPTY;
+        }
+    }
+
+    private String resolveCommit(final SCMRevision revision) {
+        if (revision instanceof AbstractGitSCMSource.SCMRevisionImpl) {
+            return ((AbstractGitSCMSource.SCMRevisionImpl) revision).getHash();
         }
-        else if (revision.get() instanceof PullRequestSCMRevision) {
-            return ((PullRequestSCMRevision) revision.get()).getPullHash();
+        else if (revision instanceof PullRequestSCMRevision) {
+            return ((PullRequestSCMRevision) revision).getPullHash();
         }
         else {
-            throw new IllegalStateException("Unsupported revision type: " + revision.get().getClass().getName());
+            throw new IllegalStateException("Unsupported revision type: " + revision.getClass().getName());
```

Comment:
Shouldn't we catch that error before we create the publisher and return then a `NullPublisher`?